Public/Src/Engine/Processes/FileAccessManifest.cs

```diff
@@ -448,7 +457,44 @@ public void AddPath(AbsolutePath path, FileAccessPolicy mask, FileAccessPolicy v
 
             m_rootNode.AddNodeWithScope(this, path, new FileAccessScope(mask, values), expectedUsn ?? ReportedFileAccess.NoUsn);
         }
+
+        private void WriteAnyBuildShimBlock(BinaryWriter writer)
+        {
+#if DEBUG
+            writer.Write(0xABCDEF04); // "ABCDEF04"
+#endif
+
+            string substituteProcessExecShimPath;
+            uint shimAllProcessesFlagValue;
+            uint numShimProcessMatches;
+            if (AnyBuildShimInfo != null)
+            {
+                substituteProcessExecShimPath = AnyBuildShimInfo.SubstituteProcessExecutionShimPath.ToString(m_pathTable);
+                shimAllProcessesFlagValue = AnyBuildShimInfo.ShimAllProcesses ? 1 : 0;
+                numShimProcessMatches = (uint)AnyBuildShimInfo.ShimProcessMatches.Count;
+            }
+            else
+            {
+                substituteProcessExecShimPath = null;
+                shimAllProcessesFlagValue = 0;
+                numShimProcessMatches = 0;
+            }
+
+            WriteChars(writer, substituteProcessExecShimPath);
+            writer.Write(shimAllProcessesFlagValue);
+            writer.Write(numShimProcessMatches);
+

```

Comment:
Shouldn't we just emit a single bit to indicate if we are going to write this tripple? (or overload null for shimPath and then skip the extra fields? Now every single process that ever gets executed under our detours is duplicating this extra bytes when there are no shims...

Reply:
Updated to predicate the full body on whether the shim path is non-empty.